src/PasswordConstraintBase.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+namespace Drupal\password_enhancements;
+
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Plugin\PluginBase;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides a base class for password constraints.
+ *
+ * @see \Drupal\password_enhancements\Annotation\PasswordConstraint
+ * @see \Drupal\password_enhancements\PasswordConstraintInterface
+ * @see \Drupal\password_enhancements\PasswordConstraintManager
+ * @see plugin_api
+ */
+abstract class PasswordConstraintBase extends PluginBase implements PasswordConstraintInterface, ContainerFactoryPluginInterface {
+
+  /**
+   * The password constraint ID.
+   *
+   * @var string
+   */
+  protected $uuid;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+
+    $this->setConfiguration($configuration);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getSummary(): array {
+    return [
+      '#markup' => '',
+      '#constraint' => [
+        'id' => $this->pluginDefinition['id'],
+        'label' => $this->name(),
+        'description' => $this->pluginDefinition['description'],
+      ],
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function name(): string {
+    return $this->pluginDefinition['name'];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getUuid(): string {
+    return $this->uuid;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getConfiguration(): array {
+    return [
+      'uuid' => $this->getUuid(),
+      'id' => $this->getPluginId(),
+      'data' => $this->configuration,
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setConfiguration(array $configuration): self {
+    $configuration += [
```

Comment:
I guess when this method gets called (unless it is called in the constructor that should be avoided) the array must contain these array keys otherwise it is unexpected behavior, isn't it?